### PR TITLE
Add the hint functionality into the solo mode

### PR DESCRIPTION
**Pull request by Chroq**
## Description

Trello: https://trello.com/c/SV9h9qBw/5-1-as-norbert-i-want-tips-on-where-i-should-play

In this PR, I have add a button to call the hint function.

Based on branch **minimax_implementation**

NB: Auto-refresh when game is won not working. 

Peek 23-04-2020 21-52

### CODE REVIEW

File: components/panels/SidePanel.js [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-        </Flex>
-      ) : null}
+      <Flex
+        alignItems="flex-end"
+        justify="center"
+        flexWrap="wrap"
+        marginTop="50%"
+        width="100%"
+      >
+        {!winner ? (
```
Comment (mchaffotte) [inline components/panels/SidePanel.js:24]: you can ease these expressions

File: pages/board/offline.js [hunk 2]
```diff
@@ -40,6 +48,16 @@ export default function OfflineBoardPage() {
     }
   };
 
+  const onHintAsked = () => {
+    if (game.winner === NO_PLAYER_VALUE) {
+      const hint = game;
+      hint.grid = cleanHints(game.grid);
```
Comment (mchaffotte) [inline pages/board/offline.js:54]: I do not understand why you clean then get an hint ... you should as well copy and not mutate the variable

File: engine/game.js [hunk 3]
```diff
@@ -9,12 +9,24 @@ import {
...
 var dijkstra = require("dijkstrajs");
 var find_path = dijkstra.find_path;
 
+/**
+ * Remove all hint values presents in the grid.
+ *
+ * @param {Array} grid
+ */
+export const cleanHints = (grid) => {
+  return grid.map((value) => {
```
Comment (mchaffotte) [inline engine/game.js:25]: the readability can be better by removing return when you have only one statement

File: components/panels/SidePanel.js (hunk 1 above)
Comment (Chroq) [reply components/panels/SidePanel.js:24]: Fixed

**Pull request merged**